code/modules/admin/force_event.dm

```diff
@@ -0,0 +1,68 @@
+///Allows an admin to force an event
+/client/proc/forceEvent()
+    set name = "Trigger Event"
+    set category = "Admin.Events"
+
+    if(!holder ||!check_rights(R_FUN))
+        return
+
+    holder.forceEvent()
+
+///Opens up the Force Event Panel
+/datum/admins/proc/forceEvent()
+    if(!check_rights(R_FUN))
+        return
+
+    var/datum/force_event/ui = new(usr)
+    ui.ui_interact(usr)
+
+/// Force Event Panel
+/datum/force_event
+
+/datum/force_event/ui_interact(mob/user, datum/tgui/ui)
+    ui = SStgui.try_update_ui(user, src, ui)
+    if(!ui)
+        ui = new(user, src, "ForceEvent")
+        ui.open()
+
+/datum/force_event/ui_state(mob/user)
+    return GLOB.admin_state
+
+/datum/force_event/ui_static_data(mob/user)
+    var/list/data = list()
+    data["categories"] = list()
+    for(var/datum/round_event_control/event_control in SSevents.control)
+        if(!data["categories"][event_control.category])
+            data["categories"][event_control.category] = list(
+                "name" = event_control.category,
+                "events" = list()
+            )
+        data["categories"][event_control.category]["events"] += list(list(
+            "name" = event_control.name,
+            "description" = event_control.description,
+            "type" = event_control.type
+        ))
+    return data
+
+/datum/force_event/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+    if(..())
+        return
+    if(!check_rights(R_FUN))
+        return
+    switch(action)
+        if("forceevent")
+            var/announce_event = params["announce"]
+            var/string_path = params["type"]
+            if(!string_path)
+                return
+            var/event_to_run_type = text2path(string_path)
+            if(!event_to_run_type)
+                return
+            var/datum/round_event_control/E = locate(event_to_run_type) in SSevents.control
+            if(E)
+                E.admin_setup(usr)
+                var/datum/round_event/event = E.runEvent(announce_chance_override = announce_event ? 100 : 0)
```

Comment:
What is 100 here? Is it meant to be 10 SECONDS?

Reply:
Probability percentage. Overrides announceChance, which is used as such:```	if(activeFor == announceWhen && prob(announceChance))
		processing = FALSE
		announce(FALSE)
		processing = TRUE```

Reply:
I could alternatively replace announce_chance_override with a three stat define, saying ANNOUNCE_CHANCE_USE_DEFAULT, ANNOUNCE_CHANCE_SILENT, and ANNOUNCE_CHANCE_ALWAY_ANNOUNCE

Reply:
In the meanwhile, I have demagiced the numbers with two vars, to better explain in code what happens there.